frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber/EditNumber.tsx

```diff
@@ -126,14 +193,67 @@ export const EditNumber = ({ field }: EditNumberProps): JSX.Element => {
         message: 'Cannot be more than 10000',
       },
     }),
+    [],
+  )
+
+  // We use the customMin field to perform cross-field validation for
+  // the number range
+  const RangeMinimumValidationOptions: RegisterOptions<
+    EditNumberInputs,
+    'ValidationOptions.RangeValidationOptions.customMin'
+  > = useMemo(
+    () => ({
+      validate: {
+        // Validate that at least one of customMin/customMax is specified
+        hasRange: (customMin) => {
+          const customMax = getValues(
+            'ValidationOptions.RangeValidationOptions.customMax',
+          )
+          return (
+            customMax !== '' || customMin !== '' || 'Please enter range values'
+          )
+        },
+        hasValidRange: (customMin) => {
+          const customMax = getValues(
+            'ValidationOptions.RangeValidationOptions.customMax',
+          )
+
+          return (
+            customMax === '' ||
+            customMin === '' ||
+            customMin <= customMax ||
+            'Minimum must be less than maximum'
+          )
+        },
+      },
+      min: {
+        value: 1,
+        message: 'Minimum cannot be 0',
+      },
+    }),
     [getValues],
   )
 
+  const RangeMaximumValidationOptions: RegisterOptions<
+    EditNumberInputs,
+    'ValidationOptions.RangeValidationOptions.customMax'
+  > = useMemo(
+    () => ({
+      min: {
+        value: 1,
+        message: 'Maximum cannot be 0',
+      },
+    }),
+    [],
+  )
+
   // Effect to clear validation option errors when selection limit is toggled off.
```

Comment:
Should we update this comment to reflect that it is when the selected validation is cleared?

Reply:
I've updated it, let me know if the comment is clear enough!

Reply:
seems good enough to me. Thanks!